spec/lib/invoice_generator_spec.rb

```diff
@@ -28,28 +28,43 @@ def generate_billing_record(project, resource, span, amount = 5000)
     )
   end
 
-  def check_invoice_for_single_vm(invoices, project, vm, duration, begin_time)
+  def check_invoice_for_single_vm(invoices, project, vm, duration, begin_time, expected_vat_info: nil)
     expect(invoices.count).to eq(1)
-    expected_issuer = {
-      "name" => "Ubicloud Inc.",
-      "address" => "310 Santa Ana Avenue",
-      "country" => "US",
-      "city" => "San Francisco",
-      "state" => "CA",
-      "postal_code" => "94127"
-    }
+    expected_issuer = if expected_vat_info
+      {
+        "name" => "Ubicloud B.V.",
+        "address" => "Turfschip 267",
+        "country" => "NL",
+        "city" => "Amstelveen",
+        "postal_code" => "1186 XK",
+        "tax_id" => "NL864651442B01",
+        "trade_id" => "88492729",
+        "in_eu_vat" => true
+      }
+    else
+      {
+        "name" => "Ubicloud Inc.",
+        "address" => "310 Santa Ana Avenue",
+        "country" => "US",
+        "city" => "San Francisco",
+        "state" => "CA",
+        "postal_code" => "94127"
+      }
+    end
+
     expected_billing_info = project.billing_info ? {
       "id" => project.billing_info.id,
       "ubid" => project.billing_info.ubid,
-      "name" => "ACME Inc.",
-      "email" => nil,
-      "address" => "",
-      "country" => "NL",
-      "city" => nil,
-      "state" => nil,
-      "postal_code" => nil,
-      "tax_id" => "123456",
-      "company_name" => nil
+      "name" => project.billing_info.stripe_data["name"],
+      "email" => project.billing_info.stripe_data["email"],
+      "address" => project.billing_info.stripe_data["address"]["line1"],
+      "country" => project.billing_info.stripe_data["address"]["country"],
+      "city" => project.billing_info.stripe_data["address"]["city"],
+      "state" => project.billing_info.stripe_data["address"]["state"],
+      "postal_code" => project.billing_info.stripe_data["address"]["postal_code"],
+      "tax_id" => project.billing_info.stripe_data["metadata"]["tax_id"],
+      "company_name" => project.billing_info.stripe_data["metadata"]["company_name"],
```

Comment:
I recommend we assign `project.billing_info.stripe_data` to a local variable and use that to DRY this up.

Reply:
I removed `Serializers::BillingInfo` and refactor this part at https://github.com/ubicloud/ubicloud/pull/2977/commits/f847804386553b2c9faa6778e9b5e06095dd230b

Can you take a look at it?